test/controllers/auth/AuthActionSpec.scala

```diff
@@ -132,18 +165,42 @@ class AuthActionSpec extends UnitSpec with GuiceOneAppPerSuite with MockitoSugar
     }
   }
 
+  "A user with a confidence level 50 and neither SA enrolment" should {
+    "see unauthorized" in {
+      val unauthorisedRoute = routes.ErrorController.notAuthorised.url
+
+      val retrievalResult: Future[Enrolments ~ Option[String] ~ Option[Credentials] ~ Option[String]] =
+        Future.successful(
+          Enrolments(Set.empty) ~
+            Some("") ~ Some(fakeCredentials) ~ None
+        )
+
+      when(
+        mockAuthConnector
+          .authorise[Enrolments ~ Option[String] ~ Option[Credentials] ~ Option[String]](any(), any())(any(), any()))
+        .thenReturn(retrievalResult)
+
+      val authAction = new AuthActionImpl(mockAuthConnector, FakeAuthAction.mcc)
+      val controller = new Harness(authAction)
+
+      val result = controller.onPageLoad()(FakeRequest("", ""))
+      status(result) shouldBe SEE_OTHER
+      redirectLocation(result).get should endWith(unauthorisedRoute)
```

Comment:
Avoid using .get on options. This could be: 

`redirectLocation(result) shouldBe Some(controllers.routes.ErrorController.notAuthorised())`